### PR TITLE
Ceds 5606 Migrate Transport Country field

### PR DESCRIPTION
**Pull request by magicaljellybeans**

### CODE REVIEW

File: app/uk/gov/hmrc/exports/migrations/changelogs/cache/RenameAndConvertBorderTransportNationalityField.scala [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        declaration
+      }
+
+      val updatedDec = borderTransportData.map { data =>
+        countriesService.getCountryCode(data) match {
+          case Some(code) => renameFieldWithCountryCode(code)
+          case None if countriesService.allCountries.map(_.countryCode).contains(data) =>
+            renameFieldWithCountryCode(data) // Data is already a country code
+          case _ =>
+            logger.warn(s"Unable to find country code for [$data] while migrating.")
```
Comment (glidester) [inline app/uk/gov/hmrc/exports/migrations/changelogs/cache/RenameAndConvertBorderTransportNationalityField.scala:80]: This is something we should look for in the logs when we deploy into each environment to check for bad values

File: test/it/uk/gov/hmrc/exports/migrations/changelogs/cache/RenameAndConvertBorderTransportNationalityFieldISpec.scala [hunk 2]
```diff
@@ -0,0 +1,1185 @@
...
+    }
+
+    "rename countryName field but leave contents when already a country code " in {
+      runTest(declarationBeforeMigration2, declarationAfterMigration)
+    }
+
+    "not alter documents already migrated" in {
+      runTest(declarationAfterMigration, declarationAfterMigration)
+    }
+  }
```
Comment (glidester) [inline test/it/uk/gov/hmrc/exports/migrations/changelogs/cache/RenameAndConvertBorderTransportNationalityFieldISpec.scala:28]: Need to test the 'not migrated but containing a none-mapable country value' scenario

**Review by glidester — CHANGES_REQUESTED** [review]

File: test/it/uk/gov/hmrc/exports/migrations/changelogs/cache/RenameAndConvertBorderTransportNationalityFieldISpec.scala (hunk 2 above)
Comment (magicaljellybeans) [reply test/it/uk/gov/hmrc/exports/migrations/changelogs/cache/RenameAndConvertBorderTransportNationalityFieldISpec.scala:31]: Added

**Review by Dweik97 — DISMISSED** [review]

**Review by glidester — DISMISSED** [review]

**Review by udinobi — APPROVED** [review]
